agent/eni/watcher/watcher_windows.go

```diff
@@ -0,0 +1,168 @@
+	agentState           dockerstate.TaskEngineState
+	eniChangeEvent       chan<- statechange.Event
+	primaryMAC           string
+	interfaceMonitor     iphelperwrapper.InterfaceMonitor
+	notifications        chan int
+	netutils             networkutils.NetworkUtils
+}
+
+// NewWindowsWatcher is used to return an instance of the ENIWatcher
+// This method starts the interface monitor as well.
+// If the interface monitor cannot be started then we return an error
+// newWatcher is used to nest the return of the ENIWatcher struct
+func newWatcher(ctx context.Context,
+	primaryMAC string,
+	state dockerstate.TaskEngineState,
+	stateChangeEvents chan<- statechange.Event) (*ENIWatcher, error) {
+
+	derivedContext, cancel := context.WithCancel(ctx)
+
+	eniMonitor := iphelperwrapper.NewMonitor()
+	notificationChannel := make(chan int)
+	err := eniMonitor.Start(notificationChannel)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error occurred while instantiating watcher")
+	}
+
+	return &ENIWatcher{
+		ctx:              derivedContext,
+		cancel:           cancel,
+		agentState:       state,
+		eniChangeEvent:   stateChangeEvents,
+		primaryMAC:       primaryMAC,
+		interfaceMonitor: eniMonitor,
+		notifications:    notificationChannel,
+		netutils:         networkutils.New(),
+	}, nil
+}
+
+// reconcileOnce is used to reconcile the state of the ENIs attached to the instance
+// We get a map of all the interfaces and then reconcile the state of each interface
+func (eniWatcher *ENIWatcher) reconcileOnce() error {
+	state, err := eniWatcher.getAllInterfaces()
+	if err != nil {
+		return err
+	} else if state == nil || len(state) == 0 {
+		return nil
+	}
+
+	for mac := range state {
+		if err := eniWatcher.sendENIStateChange(mac); err != nil {
+			// This is not an error condition as the eni status message has already been sent
+			if strings.Contains(err.Error(), eniStatusSentMsg) {
+				continue
+			} else if _, ok := err.(*unmanagedENIError); ok {
+				log.Debugf("Windows ENI watcher reconciliation: unable to send state change: %v", err)
+			} else {
+				log.Warnf("Windows ENI watcher reconciliation: unable to send state change: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+// eventHandler is used to continuously monitor the notification channel for any notification
+// If any new notification is received then we will retrieve the MAC address of that interface and send the ENI state change event
+// If the context is cancelled then we will stop the monitor
+func (eniWatcher *ENIWatcher) eventHandler() {
+	for {
+		select {
+		case event := <-eniWatcher.notifications:
+			if event == iphelperwrapper.InitialNotification {
+				continue
+			}
+			go func(ctx context.Context, timeout time.Duration) {
+				mac, err := eniWatcher.netutils.GetInterfaceMACByIndex(event, ctx, timeout)
+				if err != nil {
+					log.Debugf("Windows ENI Watcher : event-handler : Cannot retrieve MAC Address for the interface with index : %v", event)
```

Comment:
You do not need extra space before the `:`. 

Please update it across this file. 

Please refrain from using mixed case messages. 

Update log like `"Unable to obtain mac address from event: %v"`

Reply:
Updated the logs to pattern "windows eni watcher event-handler: unable to obtain Mac address for interface index %v"

The prefix has been added to emphasize that the error happened during event-handler call as opposed to reconciliation call.